packages/venia-concept/src/components/Checkout/braintreeDropin.js

```diff
@@ -0,0 +1,84 @@
+/**
+ * @fileoverview This component uses the BrainTree Web Drop In to
+ * hook into Web Payments and the Payment Request API to
+ * submit payments via BrainTree.
+ *
+ * See:
+ * https://github.com/braintree/braintree-web-drop-in
+ * https://braintree.github.io/braintree-web-drop-in/docs/current/index.html
+ * https://developers.braintreepayments.com/guides/drop-in/overview/javascript/v3
+ */
+
+import React, { Component } from 'react';
+import { bool, func } from 'prop-types';
+import dropin from 'braintree-web-drop-in';
+
+const { BRAINTREE_TOKEN } = process.env;
+
+class BraintreeDropin extends Component {
+    static propTypes = {
+        isRequestingPaymentNonce: bool,
+        onError: func.isRequired,
+        onSuccess: func.isRequired
+    };
+
+    async componentDidMount() {
+        this.braintree = await this.createDropinInstance();
+    }
+
+    componentDidUpdate(prevProps) {
+        if (
+            this.braintree &&
+            this.props.isRequestingPaymentNonce &&
+            !prevProps.isRequestingPaymentNonce
+        ) {
+            // Our parent is telling us to request the payment nonce.
+            this.requestPaymentNonce();
+        }
+    }
+
+    render() {
+        return <div id="dropin-container" />;
```

Comment:
We should probably use an `id` that is more specific to `braintree` here, I would think.